libcontainer/container_linux.go

```diff
@@ -229,6 +239,25 @@ func (c *Container) Exec() error {
 	return c.exec()
 }
 
+// terminate is to kill the container's init/exec process when got failure.
+func (c *Container) terminate(process *Process) {
+	if process.ops != nil {
+		if err := ignoreTerminateErrors(process.ops.terminate()); err != nil {
+			if process.Init {
+				logrus.WithError(err).Warn("unable to terminate initProcess")
+			} else {
+				logrus.WithError(err).Warn("unable to terminate setnsProcess")
+			}
+		}
+		if process.Init {
+			_ = c.cgroupManager.Destroy()
+			if c.intelRdtManager != nil {
+				_ = c.intelRdtManager.Destroy()
+			}
```

Comment:
As we're logging errors above, should these both also log a warning when failing to destroy? (perhaps specific errors can be ignored)

Reply:
It's really more readable, I think. Thanks.